### ISSUE TITLE
Report if Neo4j Neo4j store has not been upgraded

### ISSUE THREAD

**Issue by jacontre-c** [CiscoDevNet/sure#6]
For 20.3, we should see v0.A.9. If the returned store version is older:

2021-08-21 18:22:21.909+0000 INFO [o.n.i.d.DiagnosticsManager]   NeoStore v0.A.8

It may point to a previous upgrade failure.

This scenario will cause 20.6 upgrade failures:
org.neo4j.kernel.impl.storemigration.StoreUpgrader$UnexpectedUpgradingStoreVersionException: Not possible to upgrade a store with version 'v0.A.8' to current store version `SF4.0.0` (Neo4j 4.1.7).
 

This needs manual recovery

**Comment by rukapse** [comment]
ADD INFO:

in each neo4j release there seems to be like a record format with its own version stamp which works within
that specific release. Our customer is using software release 20.3.2 which has neo4j binaries of version
3.5.14. This gets printed in debug.log :
 
2021-08-21 18:22:21.879+0000 INFO [o.n.i.d.DiagnosticsManager] Kernel version: 3.5.14,932d555a90a30c92acddd30a504e6ae1932a92ae
 
The record/store version can also be seen in the same log :
 
2021-08-21 18:22:21.909+0000 INFO [o.n.i.d.DiagnosticsManager]   NeoStore v0.A.8
 
When checking other customers running same vManage release 20.3.2, they also have binaries of release 3.5.14, but their
stores are at version v0.A.9. Somehow, the journey of getting the customer to 20.3.2 did not not get their store to
v0.A.9. Yet : the whole vManage works fine with this release but when upgrading to 20.6.1 the upgrade fails with error as follows :
 
org.neo4j.kernel.impl.storemigration.StoreUpgrader$UnexpectedUpgradingStoreVersionException: Not possible to upgrade a store with version 'v0.A.8' to current store version `SF4.0.0` (Neo4j 4.1.7).
        at org.neo4j.kernel.impl.storemigration.StoreUpgrader.getVersionFromResult(StoreUpgrader.java:239)
        at org.neo4j.kernel.impl.storemigration.StoreUpgrader.migrate(StoreUpgrader.java:202)
        at org.neo4j.kernel.impl.storemigration.StoreUpgrader.migrateIfNeeded(StoreUpgrader.java:153)   
 
The new neo4j binaries cannot handle the old records and we are stuck with our upgrade.
 
Not sure how widespread this issue is. Probably just a corner case, but maybe it makes sense to get some matrix or table
to verify whether the neo4j binaries are running with the most current store version so we are not failing upgrade?
 
The problem we ran into is described in this neo4j article :
 
https://community.neo4j.com/t/upgrading-from-3-5-8-to-4-0-3-fails-for-existing-database/16520/2
 
We have a BEMS opened for then problem : https://clpsvs.cloudapps.cisco.com/services/clip/main/app/service/bems/engagement/724427608/engagement
 
I am inclined to execute DB upgrade within release 20.3.2 hoping it would bring the record format to v0.A.9?

**Comment by rukapse** [comment]
I struggled with this issue for quite some time where we could not do upgrade.
 
2021-10-01 11:15:46.451+0000 WARN [o.n.k.d.Database] [neo4j] Exception occurred while starting the database. Trying to stop already started components. Not possible to upgrade a store with version 'v0.A.8' to current store version `SF4.0.0` (Neo4j 4.1.7).
org.neo4j.kernel.impl.storemigration.StoreUpgrader$UnexpectedUpgradingStoreVersionException: Not possible to upgrade a store with version 'v0.A.8' to current store version `SF4.0.0` (Neo4j 4.1.7).
 
The error was explained also in this link https://community.neo4j.com/t/upgrading-from-3-5-8-to-4-0-3-fails-for-existing-database/16520/2
 
I looked at some other customer running 20.3.2 and their records were already one version higher
 
vManage-UX_Factories-1:/var/log/nms$ grep v0.A.9 debug.log
2021-07-25 07:31:31.840+0000 INFO [o.n.i.d.DiagnosticsManager]   ArrayPropertyStore v0.A.9
2021-07-25 07:31:31.840+0000 INFO [o.n.i.d.DiagnosticsManager]   NodeStore v0.A.9
2021-07-25 07:31:31.840+0000 INFO [o.n.i.d.DiagnosticsManager]   StringPropertyStore v0.A.9
2021-07-25 07:31:31.840+0000 INFO [o.n.i.d.DiagnosticsManager]   PropertyIndexStore v0.A.9
2021-07-25 07:31:31.840+0000 INFO [o.n.i.d.DiagnosticsManager]   StringPropertyStore v0.A.9
 
 
We just executed the command "request nms configuration-db upgrade` and this made the records move to 'v0.A.9'. Then we could continue
with upgrade to from 20.3.2 towards 20.6.1.
 
The upgrade for standalone is not something we do, but here the script moved he records from 'v0.A.8' to ‘v0.A.9’.
 
If we could have matrix with neo4j version and the expected store version, sure could flag this.
 
https://neo4j.com/developer/kb/store-format-versions/
 
Table

Description automatically generated
 
Graphical user interface, application

Description automatically generated
 
 
https://neo4j.com/docs/operations-manual/current/tools/neo4j-admin/neo4j-admin-store-info/
